### PR TITLE
Misc cleanups

### PR DESCRIPTION
**Pull request by rustyrussell**

### CODE REVIEW

File: lightningd/log.c [hunk 1]
```diff
@@ -18,6 +18,10 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#if DEVELOPER
+bool dev_no_backtrace;
```
Comment (cdecker) [inline lightningd/log.c]: Maybe we could use [`RUNNING_ON_VALGRIND`](http://valgrind.org/docs/manual/manual-core-adv.html#manual-core-adv.clientreq) to disable when needed (with the downside of requiring `valgrind.h` during compile time).

File: wallet/wallet.c [hunk 2]
```diff
@@ -73,7 +73,7 @@ bool wallet_update_output_status(struct wallet *w,
 	} else {
 		db_exec(__func__, w->db,
 			"UPDATE outputs SET status=%d WHERE "
-			"AND prev_out_tx = '%s' AND prev_out_index = "
+			"prev_out_tx = '%s' AND prev_out_index = "
```
Comment (cdecker) [inline wallet/wallet.c]: My bad, sorry

**Comment by cdecker** [comment]
Travis complains about an unguarded instance of `dev_no_backtrace` in log:

```
lightningd/log.c:479:7: error: 'dev_no_backtrace' undeclared (first use in this function)
  if (!dev_no_backtrace)
```

Should be easy enough to fix

File: lightningd/log.c (hunk 1 above)
Comment (rustyrussell) [reply lightningd/log.c]: Yes, let's do that.  It's simpler, and we already have HAVE_VALGRIND_MEMCHECK_H...

Comment (rustyrussell) [reply lightningd/log.c]: Uh, no... if they don't have the valgrind headers installed, but have valgrind, then "make check" will fail.  We need this :(

**Pull request merged**